halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/DeploymentEnvironment.java

```diff
@@ -94,6 +94,7 @@ public static Size fromString(String name) {
   private String accountName;
   private Consul consul = new Consul();
   private Vault vault = new Vault();
+  private String location = "spinnaker";
```

Comment:
Yes exactly. You will only need to give this treatment to k8s for now, the other cloud providers won't call `getLocation()` until they've been given the same refactor as you have provided here. I would probably have the semantics be defined in `getLocation()` which only looks up a statically defined method on the cloud provider of the account used to deploy to when `String location` is null (hasn't been user defined yet).